src/main/java/org/launchcode/water_garden_tour/controllers/TourController.java

```diff
@@ -0,0 +1,70 @@
+package org.launchcode.water_garden_tour.controllers;
+
+
+import org.apache.tomcat.util.http.fileupload.IOUtils;
+import org.launchcode.water_garden_tour.models.data.GardenRepository;
+import org.launchcode.water_garden_tour.models.garden.Garden;
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Controller
+@RequestMapping("user")
+public class TourController {
+
+    @Autowired
+    private GardenRepository gardenRepository;
+
+    List<Garden> tourGardens = new ArrayList<>();
+
+    @GetMapping("/gardens/tour/{gardenId}")
+    public String addTourGarden(Model model, @PathVariable("gardenId") Integer gardenId) {
+
+        Optional<Garden> chosenGarden = gardenRepository.findById(gardenId);
+        if (chosenGarden.isPresent()) {
+            Garden tourGarden = chosenGarden.get();
+            if (!tourGardens.contains(tourGarden)) {
+                tourGardens.add(tourGarden);
+            }
+        }
+        model.addAttribute("gardens", gardenRepository.findAll());
+        model.addAttribute("tourGardens", tourGardens);
+        model.addAttribute("title", "My Water Garden Tour");
+        return "/gardens/list";
+    }
+
+    @GetMapping("/gardens/tour")
+    public String listGardenTour(Model model) {
```

Comment:
What is this endpoint doing? Is this endpoint just listing your current tours you have?

Reply:
If I understand the section you're asking about, it is connected to the button at the top of Garden List that takes you to My Water Garden Tour.